### DISCUSSION TITLE
Orientation relationship analysis between neighbour grains

### DISCUSSION THREAD

**Discussion by ACJustin** [mtex-toolbox/mtex#483] [Ask Anything] [answered]
Hi Guys, 

I would like to analyse the OR between neighbour phases. However, when I use the neighbors function, the output of grain_no and grain_pairs is empty. Do you have any suggestion what might be the reasons?

% Plot single grain
% Select grain of interest
grain_sel = grains(530) 
% Identification of grain neighbours incl. grains within
[grain_no,grain_pairs] = neighbors(grain_sel)
% Store ID of all grain neighbours
neighbour_id = grain_pairs(1:end,1)
neighbour_grains = grains(neighbour_id)
% Elimination of fcc grains(Phase ==1)
condition = grains(neighbour_id).phase == 1
neighbour_grains = neighbour_grains(condition)

Many thanks in advance.

Best regards, 
Justin

**Comment by kilir** [comment, marked as answer]
Hi,
if you supply only one grain to neighbors, it can't have any neighbors.

This should work:
```
mtexdata titanium
grains = ebsd.calcGrains;

% grain of interest goi;
goi = 50;

% Identification of grain neighbours incl. grains within
[grain_no,grain_pairs] = neighbors(grains)

% goi n-neighbors
grain_no(goi)

% all pairs for goi
pi = any(ismember(grain_pairs,ind),2)

% id of neighbors
neighborsid = unique(grain_pairs(pi,:))
neighborsid(neighborsid==goi)=[];
```
Cheers,
Rüdiger
